src/main/java/edu/harvard/iq/dataverse/pidproviders/doi/crossref/CrossRefDOIRegisterService.java

```diff
@@ -0,0 +1,326 @@
+        } catch (Exception e) {
+            logger.log(Level.INFO, identifier, e);
+            return false;
+        }
+        return doiExists;
+    }
+
+    public HashMap<String, String> getMetadata(String identifier) throws IOException {
+        HashMap<String, String> metadata = new HashMap<>();
+        try {
+            CrossRefRESTfullClient client = getClient();
+            String jsonMetadata = client.getMetadata(identifier.substring(identifier.indexOf(":") + 1));
+            Map<String, Object> mappedJson = new ObjectMapper().readValue(jsonMetadata, HashMap.class);
+            logger.log(Level.FINE, jsonMetadata);
+            metadata.put("_status", mappedJson.get("status").toString());
+        } catch (RuntimeException e) {
+            logger.log(Level.INFO, identifier, e);
+        }
+        return metadata;
+    }
+
+    public String reserveIdentifier(String identifier, DvObject dvObject) throws IOException {
+        logger.info("Crossref reserveIdentifier");
+        String xmlMetadata = getMetadataFromDvObject(identifier, dvObject);
+
+        CrossRefRESTfullClient client = getClient();
+        return client.postMetadata(xmlMetadata);
+    }
+
+    public void modifyIdentifier(String identifier, DvObject dvObject) throws IOException {
+        logger.info("Crossref modifyIdentifier");
+        String xmlMetadata = getMetadataFromDvObject(identifier, dvObject);
+
+        CrossRefRESTfullClient client = getClient();
+        client.postMetadata(xmlMetadata);
+    }
+
+    public String getMetadataFromDvObject(String identifier, DvObject dvObject) {
+        Dataset dataset;
+
+        if (dvObject instanceof Dataset) {
+            dataset = (Dataset) dvObject;
+        } else {
+            dataset = (Dataset) dvObject.getOwner();
+        }
+        if (dataverseService == null) {
+            dataverseService = CDI.current().select(DataverseServiceBean.class).get();
+        }
+
+        CrossRefMetadataTemplate metadataTemplate = new CrossRefMetadataTemplate();
+        metadataTemplate.setIdentifier(identifier.substring(identifier.indexOf(':') + 1));
+        metadataTemplate.setAuthors(dataset.getLatestVersion().getDatasetAuthors());
+        metadataTemplate.setDepositor(depositor);
+        metadataTemplate.setDepositorEmail(depositorEmail);
+        metadataTemplate.setInstitution(dataverseService.getRootDataverseName());
+
+        String title = dvObject.getCurrentName();
+        if (dvObject.isInstanceofDataFile()) {
+            //Note file title is not currently escaped the way the dataset title is, so adding it here.
+            title = StringEscapeUtils.escapeXml10(title);
+        }
+
+        if (title.isEmpty() || title.equals(DatasetField.NA_VALUE)) {
+            title = AbstractPidProvider.UNAVAILABLE;
+        }
+
+        metadataTemplate.setTitle(title);
+
+        return metadataTemplate.generateXML();
+    }
+}
+
+class CrossRefMetadataTemplate {
```

Comment:
Out of scope perhaps for now - we should coordinate and see if the enhancements in #10615/#10632 should be applied to CrossRef as well (as part of that PR or a new one once they're merged). It's possible that CrossRef doesn't accept all the new metadata that gets added in those, but if it does, I assume we'd want to send it to CrossRef as well. @konradperlowski - any thoughts?